src/bindings/html/dom.js

```diff
@@ -77,19 +76,31 @@ export function translateMutations(view, langs, mutations) {
 
   const elements = [];
 
-  targets.forEach(target => target.childElementCount ?
-      elements.push(...getTranslatables(target)) : elements.push(target));
+  targets.forEach(target => {
+    if (target.childElementCount) {
+      elements.push(...getTranslatables(target));
+    } else {
+      if (target.hasAttribute('data-l10n-id')) {
+        elements.push(target);
+      }
+    }
+  });
+
+  if (elements.length) {
+    translateElements(view, langs, elements);
+  }
+}
 
-  Promise.all(
-    elements.map(elem => getElementTranslation(view, langs, elem))).then(
-      translations => applyTranslations(view, elements, translations));
+function translateElements(view, langs, elements) {
+  return view.ready.then(() => {
```

Comment:
I like that we're not repeating ourselves here and that you abstracted `translateElements` into its own function.  Can you move it closer to where `translateElement` is defined?

However, I think that the `view.ctx.fetch` guard should be set in view.js, not here.  In dom.js, we make certain assumptions and one of them is that the view takes care of guaranteeing that the first language has been loaded.

(We can discuss if this assumption is valid.  For now, I suggest we keep it and keep things consistent.)

Take a look at `translateFragment` in dom.js and in view.js.  The view calls `ctx.fetch` before it calls dom's `translateFragment`.  I think we should do the same for mutations:  in view's `onMutations` we should add `ctx.fetch`, just like it's done a few lines above in `translateFragment`.

What do you think?